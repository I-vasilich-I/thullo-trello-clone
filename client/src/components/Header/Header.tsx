import Link from 'next/link';
import Image from 'next/image';
import styled from 'styled-components';
import { LINKS } from 'src/constants';
import LogoImg from 'public/images/Logo.svg';
import { font } from 'src/styles/mixins.style';
import { HEADER_HEIGHT } from 'src/styles/variables.style';

interface ILinkProps {
  signUp?: boolean;
}

const StyledHeader = styled.header`
  position: fixed;
  top: 0;
  width: 100%;
  height: ${HEADER_HEIGHT};
  background-color: ${({ theme }) => theme.colors.$white};
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.05);
`;

const HeaderWrapper = styled.div`
  max-width: 1440px;
  width: 100%;
  height: 100%;
  padding: 0 24px;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 10px;
`;

const HeaderLeft = styled.div`
  max-width: 554px;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const HeaderRight = styled.div`
  max-width: 552px;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: flex-end;
`;

const HeaderAuth = styled.div`
  display: flex;
  align-items: center;
`;

const StyledLink = styled.a<ILinkProps>`
  width: 80px;
  height: 35px;
  padding: 0.8rem 0.8rem;
  display: inline-block;
  text-decoration: none;
  text-align: center;
  white-space: nowrap;
  border: 1px solid transparent;
  border-radius: 6px;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out;

  ${({ theme }) =>
    font({
      size: '1.4rem',
      height: '1.6rem',
      color: theme.colors.$blue1,
      family: 'Noto Sans Bold',
      weight: 'bold',
      spacing: '0.035em',
    })}

  color: ${({ theme, signUp }) => (signUp ? theme.colors.$white : theme.colors.$blue1)};
  background-color: ${({ theme, signUp }) => (signUp ? theme.colors.$blue1 : 'transparent')};

  &:hover {
    background-color: ${({ theme, signUp }) => (signUp ? theme.colors.$blue2 : 'transparent')};
    text-decoration: ${({ signUp }) => (signUp ? 'none' : 'underline')};

    cursor: pointer;
  }
`;

const { SIGN_IN, SIGN_UP, HOME } = LINKS;

const Header = () => {
  return (
    <StyledHeader>
      <HeaderWrapper>
        <HeaderLeft>
          <Link href={HOME.href}>
            <a>
              <Image src={LogoImg} alt="logo" width={98} height={29} />
            </a>
          </Link>
        </HeaderLeft>
        <HeaderRight>
          <HeaderAuth>
            <Link href={SIGN_IN.href} passHref>
              <StyledLink>{SIGN_IN.title}</StyledLink>
            </Link>
            <Link href={SIGN_UP.href} passHref>
              <StyledLink signUp>{SIGN_UP.title}</StyledLink>
            </Link>
          </HeaderAuth>
        </HeaderRight>
      </HeaderWrapper>
    </StyledHeader>
  );
};

export default Header;
