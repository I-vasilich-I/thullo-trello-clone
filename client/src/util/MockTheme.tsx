import { ReactNode } from 'react';
import { theme } from 'src/styles/theme.style';
import { ThemeProvider } from 'styled-components';

interface Props {
  children: ReactNode;
}

const MockTheme: React.FC<Props> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export default MockTheme;
