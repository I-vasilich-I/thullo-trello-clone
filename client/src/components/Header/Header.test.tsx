import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import MockTheme from 'src/util/MockTheme';
import Header from './Header';

describe('Header', () => {
  let container: HTMLElement;

  beforeEach(() => {
    const elem = render(
      <MockTheme>
        <Header />
      </MockTheme>,
    );

    container = elem.container;
  });

  it('renders links', () => {
    const links = screen.getAllByRole('link');
    expect(links.length).toBe(3);
  });

  it('renders logo', () => {
    const logo = screen.getByAltText('logo');
    expect(logo).toBeInTheDocument();
  });

  it('renders header unchanged', () => {
    expect(container).toMatchSnapshot();
  });
});
