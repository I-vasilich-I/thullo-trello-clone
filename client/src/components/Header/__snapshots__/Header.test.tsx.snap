// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders header unchanged 1`] = `
<div>
  <header
    class="sc-c258c1e4-0 hnlhlF"
  >
    <div
      class="sc-c258c1e4-1 dEocDQ"
    >
      <div
        class="sc-c258c1e4-2 gCZhGs"
      >
        <a
          href="/"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2798%27%20height=%2729%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="logo"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </a>
      </div>
      <div
        class="sc-c258c1e4-3 bCeIIR"
      >
        <div
          class="sc-c258c1e4-4 hudwCh"
        >
          <a
            class="sc-c258c1e4-5 gomyFW"
            href="/signin"
          >
            Sign in
          </a>
          <a
            class="sc-c258c1e4-5 GgNRL"
            href="/signup"
          >
            Sign up
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
